Reference/google/protobuf/any_test.pb.swift

```diff
@@ -123,35 +108,42 @@ struct ProtobufUnittest_TestAny: SwiftProtobuf.Proto3Message, SwiftProtobuf._Mes
   init() {}
 
   mutating func _protoc_generated_decodeMessage<D: SwiftProtobuf.Decoder>(decoder: inout D) throws {
-    try _uniqueStorage().decodeMessage(decoder: &decoder)
+    try withExtendedLifetime(_storage) { (_storage: _StorageClass) in
+      while let fieldNumber = try decoder.nextFieldNumber() {
+        try decodeField(decoder: &decoder, fieldNumber: fieldNumber)
+      }
+    }
   }
 
   mutating func _protoc_generated_decodeField<D: SwiftProtobuf.Decoder>(decoder: inout D, fieldNumber: Int) throws {
-    try _uniqueStorage().decodeField(decoder: &decoder, fieldNumber: fieldNumber)
+    switch fieldNumber {
```

Comment:
@allevato:  I just noticed this:  How are you ensuring proper value semantics here?  You've dropped the `_uniqueStorage()` copy-on-write support for these mutating methods, so it looks like two copies of the same object now risk mutating the shared backing store.

That's clearly not a problem in the case where decoding is occurring during initialization, but now that we have `merge` methods, decoding can occur at any time.